webapp/src/features/dj/DjList.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import { collection, getDocs, query } from "firebase/firestore";
+import { Dj } from "../../util/types";
+import { db } from "../../util/firebase";
+import { Alert, AlertHeading, Breadcrumb, Button, Spinner, Stack, Table } from "react-bootstrap";
+import { Link, useNavigate } from "react-router-dom";
+import { docToRawType } from "../../store/util";
+
+type Props = {
+    past?: boolean;
+}
+
+const DjList = ({ past = false}: Props) => {
+    const [djs, setDjs] = useState<Dj[]>([]);
+    const [loading, setLoading] = useState<boolean>(false);
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        setLoading(true);
+        (async () => {
+            const q = query(collection(db, "djs"));
+            const querySnapshot = await getDocs(q);
+
+            const djs: Dj[] = querySnapshot.docs
+                .map((doc) => docToRawType<Dj>(doc))
+                .filter((doc): doc is Exclude<typeof doc, null> => doc !== null);
```

Comment:
TypeScript question: what does: 
```
doc is Exclude<typeof doc, null>
``` 
do here?

Reply:
So that's a little bit of jank made to fliter out any empty types so the typescript compiler (and technically us too) knows for sure we're not going to be putting empty records in that dj object.  It's not something we'd likely see, but something in the google firebase code says "it's not guranteed that this isn't defined", so we're just adding checks for that